hudi-client/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -77,6 +77,17 @@
   public static final String BLOOM_INDEX_INPUT_STORAGE_LEVEL = "hoodie.bloom.index.input.storage.level";
   public static final String DEFAULT_BLOOM_INDEX_INPUT_STORAGE_LEVEL = "MEMORY_AND_DISK_SER";
 
+  /**
+   * Only applies if index type is GLOBAL_BLOOM.
+   * <p>
+   * When set to true, an update including the partition path of a record that already exists will result in inserting
+   * the incoming record into the new partition and deleting the original record in the old partition.
+   * <p>
+   * When set to false, the original record will only be updated in the old partition.
+   */
+  public static final String GLOBAL_BLOOM_INDEX_SHOULD_UPDATE_PARTITION_PATH = "hoodie.global.bloom.index.should.update.partition.path";
```

Comment:
@vinothchandar : does the config param looks good. I am okay with this. just wanted your opinion.

Reply:
Since this applies to the bloom index, can we slightly rename to `hoodie.bloom.index.update.partition.path` and also shorten it a bit

Reply:
> Since this applies to the bloom index, can we slightly rename to `hoodie.bloom.index.update.partition.path` and also shorten it a bit

I like the brevity. As the docs specify as "only apply to GLOBAL_BLOOM", it is unlikely to confuse users.